modules/cover-image-workflowoperation/src/main/java/org/opencastproject/workflow/handler/coverimage/CoverImageWorkflowOperationHandlerBase.java

```diff
@@ -302,57 +304,87 @@ protected String loadXsl(WorkflowOperationInstance operation) throws WorkflowOpe
     }
   }
 
-  protected String getMetadataXml(MediaPackage mp) {
-    StaticMetadata metadata = getStaticMetadataService().getMetadata(mp);
+  protected String getMetadataXml(MediaPackage mp, WorkflowOperationInstance operation) {
+    //get specified episode/series flavor
+    String configuredEpisodeFlavor = StringUtils.trimToEmpty(operation.getConfiguration(EPISODE_FLAVOR));
+    String configuredSeriesFlavor = StringUtils.trimToEmpty(operation.getConfiguration(SERIES_FLAVOR));
 
+    //use dublincore/episode and dublincore/series metadata-catalogs as default
+    if (("").equals(configuredEpisodeFlavor)) {
+      configuredEpisodeFlavor = "dublincore/episode";
+    }
+    if (("").equals(configuredSeriesFlavor)) {
+      configuredSeriesFlavor = "dublincore/series";
+    }
+
+    String episodeFlavorType = configuredEpisodeFlavor.split("/")[0];
+    String episodeFlavorSubtype = configuredEpisodeFlavor.split("/")[1];
+    String seriesFlavorType = configuredSeriesFlavor.split("/")[0];
+    String seriesFlavorSubtype = configuredSeriesFlavor.split("/")[1];
```

Comment:
Better use

```java
MediaPackageElementFlavor.parseFlavor(flavorStr);
```
You can then access the flavor and subflavor via the flavor object's methods.
And it assures that it's a valid flavor or directly use it since you want a flavor object down below anyway.